backend/prisma/migrations/20240725045118_init_geohash_table/migration.sql

```diff
@@ -0,0 +1,40 @@
+/*
+  Warnings:
+
+  - A unique constraint covering the columns `[email]` on the table `user` will be added. If there are existing duplicate values, this will fail.
+  - Added the required column `LocationId` to the `Adoptee` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `state` to the `Adoptee` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `LocationId` to the `Adopter` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `state` to the `Adopter` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `email` to the `user` table without a default value. This is not possible if the table is not empty.
+
+*/
+-- AlterTable
+ALTER TABLE "Adoptee" ADD COLUMN     "LocationId" INTEGER NOT NULL,
+ADD COLUMN     "state" TEXT NOT NULL;
+
+-- AlterTable
+ALTER TABLE "Adopter" ADD COLUMN     "LocationId" INTEGER NOT NULL,
+ADD COLUMN     "state" TEXT NOT NULL;
+
+-- AlterTable
+ALTER TABLE "user" ADD COLUMN     "email" TEXT NOT NULL;
+
+-- CreateTable
+CREATE TABLE "Location" (
+    "id" SERIAL NOT NULL,
+    "latitude" DOUBLE PRECISION,
+    "longitude" DOUBLE PRECISION,
+    "geohash" TEXT,
+
+    CONSTRAINT "Location_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateIndex
+CREATE UNIQUE INDEX "user_email_key" ON "user"("email");
```

Comment:
Interesting, can we create an index on the geohash column too (feel free to do so in a separate commit when you have time)? to speed up querying of those in the same geohash box during gale-shapley

Reply:
The generateRandomElement function generates random geographical coordinates (latitude and longitude). Meanwhile, the generateRandomCoordinates function creates an array of these randomly generated coordinates. The encodeGeohash function encodes the latitude and longitude into a geohash. The coordinates are then recursively subdivided to ensure accurate geohashing. Finally, the geohashing process is initialized and tested.